packages/core/src/extension/loader.ts

```diff
@@ -17,21 +17,45 @@ import { EventPayloads, EventType } from "@moonlight-mod/types/core/event";
 
 const logger = new Logger("core/extension/loader");
 
-function loadExtWeb(ext: DetectedExtension) {
+function evalIIFE(id: string, source: string): ExtensionWebExports {
+  const fn = new Function("require", "module", "exports", source);
+
+  const module = { id, exports: {} };
+  fn.apply(window, [
+    () => {
+      logger.warn("Attempted to require() from web");
+    },
+    module,
+    module.exports
+  ]);
+
+  return module.exports;
+}
+
+async function evalEsm(source: string): Promise<ExtensionWebExports> {
+  // Data URLs (`data:`) don't seem to work under the CSP, but object URLs do
+  const url = URL.createObjectURL(new Blob([source], { type: "text/javascript" }));
+
+  const module = await import(url);
+
+  URL.revokeObjectURL(url);
+
+  return module;
+}
+
+async function loadExtWeb(ext: DetectedExtension) {
   if (ext.scripts.web != null) {
     const source = ext.scripts.web + `\n//# sourceURL=${ext.id}/web.js`;
-    const fn = new Function("require", "module", "exports", source);
-
-    const module = { id: ext.id, exports: {} };
-    fn.apply(window, [
-      () => {
-        logger.warn("Attempted to require() from web");
-      },
-      module,
-      module.exports
-    ]);
-
-    const exports: ExtensionWebExports = module.exports;
+
+    let exports: ExtensionWebExports;
+
+    try {
```

Comment:
I think I would rather try catch both tries and capture *both* errors, as if your extension fails to load normally, the ESM load can be a little misleading if it errors for different reasons. I'll change the log if this ever trips anyone up, but I don't think it'll do so since it would be hard to get both to fail for different reasons tmk.